src/Components/ScrivitoExtensions/ColumnsEditorTab.js

```diff
@@ -5,13 +5,17 @@ import { flatten, isEqual, last, take, takeRight, times } from "lodash-es";
 import ColumnWidget from "../../Widgets/ColumnWidget/ColumnWidgetClass";
 
 function ColumnsEditorTab({ widget }) {
-  const originalContents = React.useMemo(
-    () => widget.get("columns").map((column) => column.get("content")),
-    [widget]
-  );
+  const [originalContents, originalContentsId] = React.useMemo(() => {
+    const contents = calculateContents(widget);
+
+    return [contents, calculateContentIds(contents)];
+  }, [widget]);
+
+  const readOnly =
+    !Scrivito.canWrite() ||
+    isConcurrentEditDetected(widget, originalContentsId);
```

Comment:
* [ ] Just switching to read-only, without any indication why, will appear to the editor like the editing is broken.